crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -0,0 +1,272 @@
+                "transaction".to_string(),
+            ]
+            .to_vec(),
+        ))
+        .await
+        .expect("Could not start Devnet");
+        let req_body = Body::from(
+            json!({
+                "address": DUMMY_ADDRESS,
+                "amount": DUMMY_AMOUNT
+            })
+            .to_string(),
+        );
+        let resp = devnet_dump.post_json("/mint".into(), req_body).await.unwrap();
+        assert_eq!(resp.status(), StatusCode::OK, "Checking status of {resp:?}");
+        let mut resp_body = get_json_body(resp).await;
+        let tx_hash_value = resp_body["tx_hash"].take();
+
+        // load transaction from file and check transaction hash
+        let devnet_load = BackgroundDevnet::spawn(Some(
+            ["--dump-path".to_string(), dump_file_name.to_string()].to_vec(),
+        ))
+        .await
+        .expect("Could not start Devnet");
+        let loaded_transaction = devnet_load
+            .json_rpc_client
+            .get_transaction_by_hash(
+                FieldElement::from_hex_be(tx_hash_value.as_str().unwrap()).unwrap(),
+            )
+            .await
+            .unwrap();
+        if let starknet_rs_core::types::Transaction::Invoke(
+            starknet_rs_core::types::InvokeTransaction::V1(invoke_v1),
+        ) = loaded_transaction
+        {
+            assert_eq!(
+                invoke_v1.transaction_hash,
+                FieldElement::from_hex_be(tx_hash_value.as_str().unwrap()).unwrap()
+            );
+        } else {
+            panic!("Could not unpack the transaction from {loaded_transaction:?}");
+        }
+
+        // remove dump file after test
+        let file_path = Path::new(dump_file_name);
+        if file_path.exists() {
+            fs::remove_file(file_path).expect("Could not remove file");
+        }
+    }
+
+    #[tokio::test]
+    async fn mint_dump_on_exit_and_load() {
+        // dump after transaction
+        let dump_file_name = "dump_on_exit";
+        let devnet_dump = BackgroundDevnet::spawn(Some(
+            [
+                "--dump-path".to_string(),
+                dump_file_name.to_string(),
+                "--dump-on".to_string(),
+                "exit".to_string(),
+            ]
+            .to_vec(),
+        ))
+        .await
+        .expect("Could not start Devnet");
+        let devnet_dump_pid = devnet_dump.process.id();
+        let req_body = Body::from(
+            json!({
+                "address": DUMMY_ADDRESS,
+                "amount": DUMMY_AMOUNT
+            })
+            .to_string(),
+        );
+        let resp = devnet_dump.post_json("/mint".into(), req_body).await.unwrap();
+        assert_eq!(resp.status(), StatusCode::OK, "Checking status of {resp:?}");
+        let mut resp_body = get_json_body(resp).await;
+        let tx_hash_value = resp_body["tx_hash"].take();
+
+        // Although dump and load is a multiplatform feature and works on all systems is troublesome
```

Comment:
Tests can't become platform dependent. Utilise     #[cfg(windows)] for creating 2nd implementation for killing process on windows

Reply:
From chatgpt.

```
    #[cfg(unix)]
    {
        Command::new("kill")
            .args(&[process_id.to_string(), signal])
            .status()
    }

    #[cfg(windows)]
    {
        // On Windows, you can use a third-party library like `ctrlc` to send signals.
        // Add `ctrlc` as a dependency in your Cargo.toml.
        // https://crates.io/crates/ctrlc
        let _ = ctrlc::send_signal(process_id, ctrlc::Signal::INT);
        Ok(ExitStatus::from_exit_code(0)) // Return a dummy exit status.
    }
```

Reply:
Also a mock entrypoint can be introduced under separate testing feature. Other cases will have to be resolved via support

Reply:
Fixed in b443161ef05e6b1c2f65fd4e30e7d90460a1abc8 and b443161ef05e6b1c2f65fd4e30e7d90460a1abc8